frame/support/src/traits/tokens/fungible/hold.rs

```diff
@@ -0,0 +1,393 @@
+	/// to be released!
+	fn release(
+		reason: &Self::Reason,
+		who: &AccountId,
+		amount: Self::Balance,
+		precision: Precision,
+	) -> Result<Self::Balance, DispatchError> {
+		// NOTE: This doesn't change the total balance of the account so there's no need to
+		// check liquidity.
+
+		// We want to make sure we can deposit the amount in advance. If we can't then something is
+		// very wrong.
+		ensure!(Self::can_deposit(who, amount, Extant) == Success, TokenError::CannotCreate);
+		// Get the amount we can actually take from the hold. This might be less than what we want
+		// if we're only doing a best-effort.
+		let amount = Self::decrease_balance_on_hold(reason, who, amount, precision)?;
+		// Increase the main balance by what we took. We always do a best-effort here because we
+		// already checked that we can deposit before.
+		let actual = Self::increase_balance(who, amount, BestEffort)?;
+		Self::done_release(reason, who, actual);
+		Ok(actual)
+	}
+
+	/// Attempt to decrease the balance of `who` which is held for the given `reason` by `amount`.
+	///
+	/// If `precision` is `BestEffort`, then as much as possible is reduced, up to `amount`, and the
+	/// amount of tokens reduced is returned. Otherwise, if the total amount can be reduced, then it
+	/// is and the amount returned, and if not, then nothing changes and `Err` is returned.
+	///
+	/// If `force` is `Force`, then locks/freezes will be ignored. This should only be used when
+	/// conducting slashing or other activity which materially disadvantages the account holder
+	/// since it could provide a means of circumventing freezes.
+	fn burn_held(
+		reason: &Self::Reason,
+		who: &AccountId,
+		mut amount: Self::Balance,
+		precision: Precision,
+		force: Fortitude,
+	) -> Result<Self::Balance, DispatchError> {
+		// We must check total-balance requirements if `!force`.
+		let liquid = Self::reducible_total_balance_on_hold(who, force);
+		if let BestEffort = precision {
+			amount = amount.min(liquid);
+		} else {
+			ensure!(amount <= liquid, TokenError::Frozen);
+		}
+		let amount = Self::decrease_balance_on_hold(reason, who, amount, precision)?;
+		Self::set_total_issuance(Self::total_issuance().saturating_sub(amount));
+		Self::done_burn_held(reason, who, amount);
+		Ok(amount)
+	}
+
+	/// Transfer held funds into a destination account.
+	///
+	/// If `on_hold` is `true`, then the destination account must already exist and the assets
+	/// transferred will still be on hold in the destination account. If not, then the destination
+	/// account need not already exist, but must be creatable.
+	///
+	/// If `precision` is `BestEffort`, then an amount less than `amount` may be transferred without
+	/// error.
+	///
+	/// If `force` is `Force`, then other fund-locking mechanisms may be disregarded. It should be
+	/// left as `Regular` in most circumstances, but when you want the same power as a `slash`, it
+	/// may be `Force`.
```

Comment:
`force` (`Fortitude`) now only has possible values `Force` and `Polite`. `Regular` is now `Polite`
```suggestion
	/// If `force` is `Force`, then other fund-locking mechanisms may be disregarded. It should be
	/// left as `Polite` in most circumstances, but when you want the same power as a `slash`, it
	/// may be `Force`.
```